src/braket/circuits/circuit.py

```diff
@@ -268,11 +268,10 @@ def _add_to_qubit_observable_mapping(self, result_type: ResultType) -> None:
             )
             new_observable = tensor_product_dict[i][0] if tensor_product_dict else observable
             current_observable = all_qubits_observable or self._qubit_observable_mapping.get(target)
-            if current_observable and current_observable != new_observable:
-                raise ValueError(
-                    f"Existing result type for observable {current_observable} for target {target}"
-                    f" conflicts with observable {new_observable} for new result type"
-                )
+
+            add_observable = Circuit._validate_observable_to_add_for_qubit(
+                current_observable, new_observable, target
+            )
 
             if result_type.target:
                 if new_observable.qubit_count > 1:
```

Comment:
I guess since this clause is unchanged, identity observables only work when they don't conflict with multi-qubit observables. Can you clarify this in the description?

Reply:
Sure thing.

Reply:
You'll probably also need to update the error message in line 290. Or if you feel like it, you could try to update the logic to make sure it works for multi-qubit observables as well. In any case, if you don't fix it in this PR, also leave a comment as a reminder in the code.

Reply:
I went ahead and updated the logic to allow for allow identities with multi-qubit observables

Reply:
If it's not too much work, can you also add a test case covering this logic?

Reply:
Yep, I included that in the commit, lines 654 and 659 in the test file